ompi/proc/proc.h

```diff
@@ -369,16 +369,42 @@ static inline bool ompi_proc_is_sentinel (ompi_proc_t *proc)
     return (intptr_t) proc & 0x1;
 }
 
-static inline intptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
+/*
+ * we assume an ompi_proc_t is at least aligned on two bytes,
+ * so if the LSB of a pointer to an ompi_proc_t is 1, we have to handle
+ * this pointer as a sentinel instead of a pointer.
+ * a sentinel can be seen as a 64 bits array with the following format :
+ * - bit  0     : 1
+ * - bits 1-15  : local jobid
+ * - bits 16-31 : job family
+ * - bits 32-63 : vpid
```

Comment:
Short [code](https://gist.github.com/bosilca/64843961946319497da7) to show the contents of the different fields. Looking at the bit position of different fields we have:
```
Operation              Raw Memory          Little endian (uint64_t)
Alterning bits  0:15   ffff000000000000                ffff
Alterning bits 15:31   0000ffff00000000            ffff0000
Alterning bits 32:63   00000000ffffffff    ffffffff00000000
Setting jobid          ffff000000000000                ffff
Setting family         0000ffff00000000            ffff0000
Setting vpid           00000000ffffffff    ffffffff00000000
```

Because opal_process_name_t is a structure, the location of the 2 fields (jobid, vpid) will never change, their offsets will always be 0 for the jobid and 4 for the vpid. Thus to your question regarding the endianness, the answer is no, you always flip the same logical bit.